### PR TITLE
Add LogHunter - a generic grep over log capability

### PR DESCRIPTION
**Pull request by mgreen27**
Add simple log grepper. Included Linux targeting and default exclusions.

### CODE REVIEW

File: artifacts/definitions/Linux/Sys/LogHunter.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    type: regex
+  - name: ExcludePathRegex
+    description: "Regex of paths to exclude."
+    default: '^/(proc|sys|run|snap)|\.journal$'
+    type: regex
+    
+sources:
+  - query: |
+      LET files = SELECT OSPath FROM glob(globs=TargetFiles,nosymlink=True)
+        WHERE NOT if(condition=ExcludePathRegex,
```
Comment (scudette) [inline artifacts/definitions/Linux/Sys/LogHunter.yaml:31]: This does not exclude the paths. We need to add a recursion callback

Comment (mgreen27) [reply artifacts/definitions/Linux/Sys/LogHunter.yaml:31]: OK no worries: Ill add recursion_callback=RecursionCallback

File: artifacts/definitions/Linux/Sys/LogHunter.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  - name: ExcludePathRegex
+    description: "Regex of paths to exclude."
+    default: '\.journal$'
+    type: regex
+    
+sources:
+  - query: |
+      LET files = SELECT OSPath 
+        FROM glob(globs=TargetFiles,nosymlink=True,
+            recursion_callback="x=> NOT x.OSPath =~ '^/(shared|proc|snap)'")
```
Comment (scudette) [inline artifacts/definitions/Linux/Sys/LogHunter.yaml:32]: This should probably be a parameter so we can exclude any nfs mounts etc

Comment (mgreen27) [reply artifacts/definitions/Linux/Sys/LogHunter.yaml:32]: the only issue I had with a parameter was if the user removes all and submits a blank. We probably need to have an if there too and I wasnt sure how that works in recursion_callback

**Pull request merged**